core/src/main/java/jenkins/cli/StopBuildsCommand.java

```diff
@@ -0,0 +1,117 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.cli;
+
+import com.google.common.annotations.VisibleForTesting;
+import hudson.Extension;
+import hudson.cli.CLICommand;
+import hudson.model.Executor;
+import hudson.model.Item;
+import hudson.model.Job;
+import hudson.model.Run;
+import jenkins.model.Jenkins;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.args4j.Argument;
+
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
+@Extension
+@Restricted(NoExternalUse.class)
+public class StopBuildsCommand extends CLICommand {
+
+    @VisibleForTesting
+    @Argument(usage = "Name of the job(s) to stop", required = true, multiValued = true)
+    /*private */ List<String> jobNames;
+
+    private boolean isAnyBuildStopped;
+
+    @Override
+    public String getShortDescription() {
+        return "Stop all running builds for job(s)";
+    }
+
+    @Override
+    protected int run() throws Exception {
+        Jenkins jenkins = Jenkins.get();
+        final HashSet<String> names = new HashSet<>();
+        names.addAll(jobNames);
+
+        final List<Job> jobsToStop = new ArrayList<>();
+        for (final String jobName : names) {
+            Item item = jenkins.getItemByFullName(jobName);
+            if (item instanceof Job) {
+                jobsToStop.add((Job) item);
+            } else {
+                throw new IllegalArgumentException("Invalid job name = " + jobName);
+            }
+        }
+
+        for (final Job job : jobsToStop) {
+            stopJobBuilds(job, job.getName());
+        }
+
+        if (!isAnyBuildStopped) {
+            stdout.println("No builds stopped;");
+        }
+
+        return 0;
+    }
+
+    private void stopJobBuilds(final Job job,
+                               final String jobName) throws IOException, ServletException {
+        final Run lastBuild = job.getLastBuild();
+        if (lastBuild != null && lastBuild.isBuilding()) {
+            stopBuild(lastBuild, jobName);
+            checkAndStopPreviousBuilds(lastBuild, jobName);
+        }
+    }
+
+    private void stopBuild(final Run build,
+                           final String jobName) throws IOException, ServletException {
+        final String buildName = build.getDisplayName();
+        Executor executor = build.getExecutor();
+        if (executor != null) {
+            isAnyBuildStopped = true;
+            executor.doStop();
```

Comment:
This does not gracefully handle the case when the current user does not have permission to stop the build. An exception will be thrown and not caught.